### DISCUSSION TITLE
nix 2.18.1

### DISCUSSION THREAD

**Discussion by dezren39** [nix-community/NixOS-WSL#311] [Q&A] [answered]
hi i use flakes and so i'm stuck needing a version 2.18.1 because my flakes all have 'lastModified' attrib in them and 2.17 won't use that. i can nix run nix/nix, for temporary uses, but i can't actually run my normal flakes or nixos build because i can't figure out how to install a different version. the file system is readonly is the error i get, or if i install it using overlay it just doesn't seem to get used.

**Comment by dezren39** [comment]
my question is how can i install a different version? or can we get the base version updated? the nix.dev script fails for readonly, and my limited understanding hasn't allowed me to force it using my configuration.nix. i can kind of understand how i could maybe do this if i upgraded to 2.17 flakes, then switched my nix package using flakes, then upgraded to my 2.18 flakes with lastModified attrib?

**Comment by K900** [comment, marked as answer]
You should be able to just `nix-shell -p nixVersions.nix_2_18`. Also, moving to discussion.
